File: wk2/pollutantmean.R
source('reader.R')

pollutantmean <- function(dir, pollutant, id = 1:332) {
    data <- readdata(dir, id)
    sums <- sapply(data, function(d) {
      readings <- !is.na(d[,pollutant])
      c(sum(d[readings, pollutant]), sum(readings))
    })
    mean <- sum(sums[1,]) / sum(sums[2,])
    round(mean, 3)
}

File: wk2/pollutantmean.jl
using DataFrames

function pollutantmean(dir, pollutant, id=1:332)
  files = [ @sprintf("%s/%03d.csv", dir, i) for i in id ]
  datasets = [ readtable(file) for file in files ]
  sums = [ sum(removeNA(getindex(data, pollutant))) for data in datasets ]
  lengths = [ length(removeNA(getindex(data, pollutant))) for data in datasets ]
  round(sum(sums) / sum(lengths), 3)
end
